### PR TITLE
feat: add code for function extract_ngram and associated tests

### LINKED ISSUE #21: Extract n-grams function and tests

### PR DESCRIPTION
**Pull request by arijeetchatterjee**
Added the following:
- Code for the function `extract_ngram`
- Tests for the above function

### CODE REVIEW

File: R/extract_ngram.R [hunk 1]
```diff
@@ -9,9 +9,37 @@
 #'
 #' @examples
 #' tweets <- c(
-#'    "Make America Great Again DonaldTrump"
+#'    "Make", "America", "Great", "Again", "DonaldTrump"
 #' )
 #' extract_ngram(tweets, n=3)
 extract_ngram <- function(tweets, n = 3) {
+  if (length(tweets) < n) {
+    stop("length of ngrams should be less than length of tweets")
```
Comment (mmaidana24318) [inline R/extract_ngram.R]: Shouldn't say "less than # of words in vector of tweets"?

Comment (arijeetchatterjee) [reply R/extract_ngram.R]: Thanks. Updated the error message.

File: R/extract_ngram.R [hunk 2]
```diff
@@ -9,9 +9,37 @@
...
+  }
 
+  if (length(tweets) < n) {
+    stop("length of ngrams should be less than length of tweets")
+  }
+  if(!is.character(tweets)) {
+    stop("input should be a character vector")
+  }
+
+  # tweets_str <- paste(tweets, sep=" ", collapse = NULL)
```
Comment (joshsia) [inline R/extract_ngram.R]: Could you please delete line 30?

**Review by joshsia — CHANGES_REQUESTED** [review]
It looks good!

File: R/extract_ngram.R (hunk 2 above)
Comment (arijeetchatterjee) [reply R/extract_ngram.R]: Sure. Missed removing the commented lines earlier. Deleted now.

**Review by joshsia — CHANGES_REQUESTED** [review]
Hi, I just noticed that we could make the tests a bit cleaner.

File: tests/testthat/test-extract_ngram.R [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+})
+test_that("Generated ngrams is incorrect", {
+  expect_equal(extract_ngram(helper_data, n=3), c("Make America Great", "America Great Again", "Great Again DonaldTrump"))
+})
+test_that("Generated ngrams is incorrect", {
+  expect_equal(extract_ngram(helper_data, n=4), c("Make America Great Again", "America Great Again DonaldTrump"))
+})
+test_that("Generated ngrams is incorrect", {
+  expect_equal(extract_ngram(helper_data, n=5), c("Make America Great Again DonaldTrump"))
+})
```
Comment (joshsia) [inline tests/testthat/test-extract_ngram.R]: I think the code will look cleaner if we group similar expectations under the same `test_that`.

```suggestion
test_that("Generated ngrams is incorrect", {
  expect_equal(extract_ngram(helper_data, n=1), c("Make", "America", "Great", "Again", "DonaldTrump"))
  expect_equal(extract_ngram(helper_data, n=3), c("Make America Great", "America Great Again", "Great Again DonaldTrump"))
  expect_equal(extract_ngram(helper_data, n=4), c("Make America Great Again", "America Great Again DonaldTrump"))
  expect_equal(extract_ngram(helper_data, n=5), c("Make America Great Again DonaldTrump"))
```

**Review by joshsia — CHANGES_REQUESTED** [review]
Continuing on from the previous comment

File: tests/testthat/test-extract_ngram.R [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Test for function extract_ngram
+helper_data <- c("Make", "America", "Great", "Again", "DonaldTrump")
+test_that("Generated ngrams is incorrect", {
+  expect_equal(extract_ngram(helper_data, n=1), c("Make", "America", "Great", "Again", "DonaldTrump"))
+  expect_equal(extract_ngram(helper_data, n=3), c("Make America Great", "America Great Again", "Great Again DonaldTrump"))
+  expect_equal(extract_ngram(helper_data, n=4), c("Make America Great Again", "America Great Again DonaldTrump"))
+  expect_equal(extract_ngram(helper_data, n=5), c("Make America Great Again DonaldTrump"))
+
```
Comment (joshsia) [inline tests/testthat/test-extract_ngram.R]: I think we're missing the closing brackets

Comment (joshsia) [inline tests/testthat/test-extract_ngram.R]:
```suggestion
test_that("Generated ngrams is incorrect", {
  expect_equal(extract_ngram(helper_data, n=1), c("Make", "America", "Great", "Again", "DonaldTrump"))
  expect_equal(extract_ngram(helper_data, n=3), c("Make America Great", "America Great Again", "Great Again DonaldTrump"))
  expect_equal(extract_ngram(helper_data, n=4), c("Make America Great Again", "America Great Again DonaldTrump"))
  expect_equal(extract_ngram(helper_data, n=5), c("Make America Great Again DonaldTrump"))
})
```

**Review by joshsia — CHANGES_REQUESTED** [review]

**Review by joshsia — APPROVED** [review]

**Review by mmaidana24318 — APPROVED** [review]

**Pull request merged**